packages/big-design/src/components/Button/spec.tsx

```diff
@@ -9,180 +10,179 @@ import { StyleableButton } from './private';
-  const { container } = render(
+  render(
     <Button variant="subtle" disabled>
       Button
     </Button>,
   );
 
-  expect(container.firstChild).toMatchSnapshot();
+  expect(screen.getByRole('button')).toMatchSnapshot();
 });
 
 test('render subtle destructive button', () => {
-  const { container } = render(
+  render(
     <Button variant="subtle" actionType="destructive">
       Button
     </Button>,
   );
 
-  expect(container.firstChild).toMatchSnapshot();
+  expect(screen.getByRole('button')).toMatchSnapshot();
 });
 
 test('render subtle destructive disabled button', () => {
-  const { container } = render(
+  render(
     <Button variant="subtle" actionType="destructive" disabled>
       Button
     </Button>,
   );
 
-  expect(container.firstChild).toMatchSnapshot();
+  expect(screen.getByRole('button')).toMatchSnapshot();
 });
 
 test('render loading button', () => {
-  const { container } = render(<Button isLoading={true}>Button</Button>);
+  render(<Button isLoading={true}>Button</Button>);
 
-  expect(container.firstChild).toMatchSnapshot();
+  expect(screen.getByRole('button')).toMatchSnapshot();
 });
 
 test('render icon only button', () => {
-  const { container } = render(<Button iconOnly={<AddIcon />}>Button</Button>);
+  render(<Button iconOnly={<AddIcon />}>Button</Button>);
 
-  expect(container.firstChild).toMatchSnapshot();
+  expect(screen.getByRole('button')).toMatchSnapshot();
 });
 
 test('render icon left button', () => {
-  const { container } = render(<Button iconLeft={<AddIcon />}>Button</Button>);
+  render(<Button iconLeft={<AddIcon />}>Button</Button>);
 
-  expect(container.firstChild).toMatchSnapshot();
+  expect(screen.getByRole('button')).toMatchSnapshot();
 });
 
 test('render icon right button', () => {
-  const { container } = render(<Button iconRight={<AddIcon />}>Button</Button>);
+  render(<Button iconRight={<AddIcon />}>Button</Button>);
 
-  expect(container.firstChild).toMatchSnapshot();
+  expect(screen.getByRole('button')).toMatchSnapshot();
 });
 
 test('render icon left and right button', () => {
-  const { container } = render(
+  render(
     <Button iconLeft={<AddIcon />} iconRight={<AddIcon />}>
       Button
     </Button>,
   );
 
-  expect(container.firstChild).toMatchSnapshot();
+  expect(screen.getByRole('button')).toMatchSnapshot();
 });
 
 test('forwards ref', () => {
   const ref = createRef<HTMLButtonElement>();
 
-  const { container } = render(<Button ref={ref} />);
-  const button = container.querySelector('button');
+  render(<Button ref={ref} />);
 
-  expect(button).toBe(ref.current);
+  expect(screen.getByRole('button')).toBe(ref.current);
 });
 
 test('triggers onClick', () => {
   const onClick = jest.fn();
 
-  const { container } = render(<Button onClick={onClick} />);
-  const button = container.firstChild as HTMLElement;
+  render(<Button onClick={onClick} />);
 
-  fireEvent.click(button);
+  fireEvent.click(screen.getByRole<HTMLButtonElement>('button'));
```

Comment:
Let's also switch out usages of `fireEvent` with `userEvent`.

Reply:
Yeah, it's a good point. I also see that we will need @testing-library/user-event package for this purpose. Is it ok to proceed?

Reply:
For sure